x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable.tsx

```diff
@@ -324,13 +325,17 @@ export class Embeddable
       this.input.onLoad(true);
     }
     const input = this.getInput();
+    const variables = {
+      searchId: generateId(),
+      ...(input.palette ? { theme: { palette: input.palette } } : {}),
+    };
```

Comment:
I'm a bit confused why the Lens embeddable needs to generate a `searchId` and pass it to the expression. Why isn't each `esaggs` expression automatically generating a searchId/inspectorId?

Reply:
> I'm a bit confused why the Lens embeddable needs to generate a `searchId` and pass it to the expression. Why isn't each `esaggs` expression automatically generating a searchId/inspectorId? This would work because the expression function is getting called each time it needs to fetch data.

That was the initial approach. But then @flash1293 found out it didn't work for brushing and subsequent requests.

Reply:
If the `esaggs` expression is generating a new one, we don't know which requests to keep in the inspector and which to kick out (keep in mind, there could be multiple `esaggs` calls in a single expression run, that's the whole issue).

The search id in this case is like a unique identifier for one "run" of the chart expression and allows us to keep everything in the inspector for the run and remove everything else.

Reply:
We could also introduce a `uuid` expression function and generate it on the expression itself:
```
set_var name="searchId" value={uuid} | build_multi_table table={esaggs searchId={var searchId} ...} table={esaggs searchId={var searchId} ...} | xy_chart
```

But I think the current approach is fine as well, no strong opinion

Reply:
@flash1293 The use case of identifying a "single run of an expression" could be something that the expression context contains, as a sibling to `variables`. I'm just not a big fan of using variables that can be overridden by canvas users.

Reply:
Right, I would be fine with that as well. @ppisljar what do you think about extending the context a liiittle bit more?

Reply:
this sounds like something that should be internal to expressions and should not leak to consumer nor the function providers. 

seems like esaggs should not `inspectorAdapters.requests?.reset();`, that should be done by expression executor on every run. seems then we don't need the search id and functions don't need to do anything, they just log requests ?

Reply:
@ppisljar Sounds like you agree that we should not be managing this from Lens. It looks like what @dej611 should do is add an `executionId` parameter to `execution.ts` which will be passed via context to `esaggs` and other functions. The inspector adapters can be _automatically reset before starting to execute.

Reply:
i am hoping functions (like esaggs) don't need to worry about this. Do we have a reason why they should ?

if executor resets inspector adapters on every run and functions just log to adapters (and don't remove anything) that should be fine right ? (so functions don't need to know about searchid)